train.py

```diff
@@ -0,0 +1,264 @@
+"""
+Finetune a pretrained gpt2 model on a custom dataset.
+    Original Paper and repository here: https://github.com/openai/gpt-2
+    Adapted from code by nshepperd: https://github.com/nshepperd/gpt-2/blob/finetuning/train.py
+"""
+import os
+import tqdm
+import time
+import argparse
+import torch
+
+from GPT2.model import GPT2LMHeadModel
+from GPT2.utils import load_weight
+from GPT2.config import get_config
+from GPT2.sample import sample_sequence
+from GPT2.encoder import get_encoder
+from GPT2.data import load_dataset, Sampler
+
+FINETUNED_DIR = 'finetuned_models'
+CHECKPOINT_DIR = os.path.join(FINETUNED_DIR, 'checkpoint')
+SAMPLE_DIR = os.path.join(FINETUNED_DIR, 'samples')
+
+
+def get_state_dict(model_name='117M'):
+    model_path = os.path.join('pretrained_models', model_name, 'model.bin')
+
+    if os.path.exists(model_path):
+        state_dict = torch.load(model_path, map_location='cpu' if not torch.cuda.is_available() else None)
+    else:
+        print('model does not exist at: {}'.format(model_path))
+        exit(0)
+    return state_dict
+
+
+def load_model(model, state_dict, device):
+    model = load_weight(model, state_dict)
+    model.to(device)
+    model.eval()
```

Comment:
@jjbrophy47  Why did you try `model.eval()`?
Didn't you mean fine tuning?

Reply:
Ooops! That line is a mistake and should be deleted, thanks.